tools/parse/src/main/scala/org/allenai/nlpstack/parse/BankerProtocol.scala

```diff
@@ -56,6 +50,91 @@ object BankerProtocol {
+    parser.parse(Sentence(Parser.tokenizeSentence(text).toIndexedSeq)) map { parse =>
+      BankerParse.fromPolytreeParse(parse)
+    }
+  }
+
+  /** Parses a sentence subject to a set of constraints.
+    *
+    * @param parser the parser to use
+    * @param sentence the sentence to parse
+    * @param constraints the constraints you want to parser to obey
+    * @return a parse tree obeying your specified constraints
+    */
+  def parseWithConstraints(
+    parser: TransitionParser,
+    sentence: Sentence,
+    constraints: Set[TransitionConstraint]
+  ): Option[BankerParse] = {
+
+    parser.parse(sentence, constraints) map { parse => BankerParse.fromPolytreeParse(parse) }
+  }
+}
+
+/** The legacy parse format used by the Banker.
+  *
+  * It has four major fields:
+  * - `tokens` is a vector of Token objects (in the order that they appear in the associated
+  * sentence). The zeroth element is assumed to be the nexus.
+  * - `breadcrumb` tells you the unique neighbor that is closer to the nexus in the
+  * undirected tree (this can be the nexus itself); for instance, if breadcrumb(5) = 3,
+  * then token 3 is one step closer to the nexus from token 5. The breadcrumb of the nexus
+  * should be -1.
+  * - `children` tells you the set of children of a node in the polytree; for instance, if
+  * children(5) = Set(3,6,7), then token 5 has three children: tokens 3, 6, and 7
+  * - `arclabels` tells you the labeled neighbors of a node in the undirected tree; for instance,
+  * if arclabels(5) = Set((4, 'det), (7, 'amod)), then token 5 has two neighbors, reached with
+  * arcs labeled 'det and 'amod (the labels are scala Symbol objects)
+  *
+  * @param sentence the parsed sentence (the zeroth token of which should be the nexus)
+  * @param breadcrumb the breadcrumb of each token (see above definition)
+  * @param children the set of children of each token in the polytree
+  * @param arclabels the set of labeled neighbors of each token in the undirected tree
+  */
+case class BankerParse(
+    sentence: Sentence,
+    breadcrumb: Vector[Int],
+    children: Vector[Set[Int]],
+    arclabels: Vector[Set[(Int, Symbol)]]
+) {
+
+  def toPolytreeParse: PolytreeParse = {
+    PolytreeParse(sentence, breadcrumb, children,
+      arclabels map { labelset =>
+        val result: Set[(Int, ArcLabel)] = labelset map {
+          case (index, label) =>
+            (index, SingleSymbolArcLabel(label))
+        }
+        result
+      })
+  }
+
+}
+
+object BankerParse {
+
+  /** Initializes a BankerParse from a PolytreeParse.
+    *
+    * @param parse the parse to initialize from
+    * @return the analogous BankerParse
+    */
+  def fromPolytreeParse(parse: PolytreeParse): BankerParse = {
+    BankerParse(parse.sentence, parse.breadcrumb, parse.children,
+      parse.arclabels map { labelset =>
+        labelset map {
+          case (index, label) =>
+            (index, label.toSymbol)
+        }
+      })
```

Comment:
Same as above, I'd put the arc label conversion on it's own line for readability (not a big deal though)